tests/test_flags/test_amend.py

```
@@ -0,0 +1,99 @@
+                                                git_commitai.main()
+
+                                                # Verify git commit --amend was called
+                                                calls = mock_run.call_args_list
+                                                assert any(["--amend" in str(call) for call in calls])
+
+    def test_amend_first_commit(self):
+        """Test --amend on the first commit (no parent)."""
+        with patch("git_commitai.run_command") as mock_run:
+            def side_effect(cmd, check=True):
+                if "git rev-parse HEAD^" in cmd:
+                    raise subprocess.CalledProcessError(1, cmd)
```

Ruff (0.12.2)

Undefined name `subprocess`

(F821)